### PR TITLE
Fix clipped render when using geometry generator symbol in layout items 

### LINKED ISSUE #58909: Layout polyline items created with the geometry generator are linked to the first existing layout map
### What is the bug or the crash?

When we create a layout polyline item and use the geometry generator to define the geometry, the geometry changes with a resizing of the first existing layout map.

### Steps to reproduce the issue

1. Create a map window that spans the layout page (Din A4)
2. Create a polyline item and use the geometry generator to define the geometry (type: LineString, units: Millimeters): 
`geom_from_wkt('POLYGON((10 10,287 10,287 200,10 200,10 10))')`
3. Resize the map window

### Versions

QGIS 3.38.2 Windows

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nyalldawson**
Fixes https://github.com/qgis/QGIS/issues/58909

### CODE REVIEW

**Review by troopa81 — COMMENTED** [review]
can we not setClipFeaturesToExtent to false for this very symbol, or set an empty extent before rendering instead of adding a new flag?

**Comment by nyalldawson** [comment]
> we not setClipFeaturesToExtent to false for this very symbol, 

That's not trivial either, we'd need to set it for all subsymbols for the symbol. The flag is less code 😁


> or set an empty extent before rendering instead of adding a new flag?

I thought about that, but I'm concerned that it might cause a new regression if someone has symbology designed around the extent being available.

File: src/core/symbology/qgssymbol.cpp [hunk 1]
```diff
@@ -1560,6 +1560,10 @@ void QgsSymbol::renderFeature( const QgsFeature &feature, QgsRenderContext &cont
       break;
     }
   }
+  if ( clippingEnabled && context.testFlag( Qgis::RenderContextFlag::DisableSymbolClippingToExtent ) )
+  {
+    clippingEnabled = false;
+  }
```
Comment (troopa81) [inline src/core/symbology/qgssymbol.cpp:1566]:
```suggestion
  clippingEnabled &= !context.testFlag( Qgis::RenderContextFlag::DisableSymbolClippingToExtent ) 
```
Cosmetic proposal. less code, better readability. We could also group all other conditions related to this boolean, but that's not mandatory.

**Review by troopa81 — APPROVED** [review]

File: src/core/symbology/qgssymbol.cpp [hunk 2]
```diff
@@ -1560,6 +1560,7 @@ void QgsSymbol::renderFeature( const QgsFeature &feature, QgsRenderContext &cont
       break;
     }
   }
+  clippingEnabled &= !context.testFlag( Qgis::RenderContextFlag::DisableSymbolClippingToExtent )
```
Comment (nyalldawson) [inline src/core/symbology/qgssymbol.cpp:1563]:
```suggestion
  clippingEnabled &= !context.testFlag( Qgis::RenderContextFlag::DisableSymbolClippingToExtent );
```

**Pull request merged**